### PR TITLE
Implement requester for Date, Time and DateTime

### LINKED ISSUE #58: Implement DateRequester
Nobody needed the DateRequester so far, but the Stub is there.

At the moment it just throws a NotImplementedException

### LINKED ISSUE #59: Implement TimeRequester
Nobody needed the TimeRequester so far, but the Stub is there.

At the moment it just throws a NotImplementedException

### PR DESCRIPTION
**Pull request by FreeApophis**
* Add first tests for requesters
* use expression bodies
* more expressions

### CODE REVIEW

File: FileManipulator/Project/MsBuild/CentralPackageVersionsManipulator.cs [hunk 1]
```diff
@@ -83,13 +83,14 @@ private static string GetVersionFromMetadata(ProjectItemElement item)
...
 
         private static string GetCentralPackagesFile(MsBuildProject project)
-            => project.GetPropertyValue(CentralPackagesFileProperty);
+            => project
+                .GetPropertyValue(CentralPackagesFileProperty);
 
         private static bool IsCentralPackageVersionsSdkImport(ResolvedImport import)
-            => import.SdkResult is { } sdk &&
-               sdk.SdkReference.Name == CentralPackageVersionsSdk;
+            => import.SdkResult is { } sdk && sdk.SdkReference.Name == CentralPackageVersionsSdk;
```
Comment (tautropfli) [inline FileManipulator/Project/MsBuild/CentralPackageVersionsManipulator.cs]: You could merge this into a single pattern I think

File: MetaGenerator/TimeKeeper.cs [hunk 2]
```diff
@@ -59,18 +59,14 @@ private void PrintMeasurements()
         }
 
         private long TotalTime()
-        {
-            return _measurements.Select(m => m.ElapsedMilliseconds).Sum();
-        }
+            => _measurements
+            .Select(m => m.ElapsedMilliseconds).Sum();
```
Comment (tautropfli) [inline MetaGenerator/TimeKeeper.cs]: *indentation

File: MetaGenerator/TimeKeeper.cs [hunk 3]
```diff
@@ -59,18 +59,14 @@ private void PrintMeasurements()
...
 
         private int EventNameColumnWidth()
-        {
-            return _measurements.Aggregate(0, (max, m) => Max(max, m.EventName.Length)) + 1;
-        }
+            => _measurements
+            .Aggregate(0, (max, m) => Max(max, m.EventName.Length)) + 1;
```
Comment (tautropfli) [inline MetaGenerator/TimeKeeper.cs]: indentation

File: MetaGenerator/UserInput/ExistingPathRequester.cs [hunk 4]
```diff
@@ -14,9 +15,7 @@ public ExistingPathRequester(IValidatedUserInput validatedUserInput)
         }
 
         protected override IEnumerable<IValidation> RequesterValidations(IUserInputDescription variable)
-        {
-            yield return new SimpleValidation(PathExists, "The path you have given does not exists, please enter an existing path:");
-        }
+            => Sequence.Return(SimpleValidation.Create(PathExists, "The path you have given does not exists, please enter an existing path:"));
```
Comment (tautropfli) [inline MetaGenerator/UserInput/ExistingPathRequester.cs]: What is the colon doing at the end?

File: MetaGenerator/UserInput/ValidatedUserInput.cs [hunk 5]
```diff
@@ -47,14 +40,22 @@ public void WriteQuestion(IUserInputDescription variable, string defaultQuestion
             _consoleWriter.WriteLine(FormatWithVariableName(question, variable.VariableName));
         }
 
+        private Func<string, Option<string>> EchoAndValidate(IUserInputDescription variable, IEnumerable<IValidation> requesterValidations)
```
Comment (tautropfli) [inline MetaGenerator/UserInput/ValidatedUserInput.cs]: Why don't you split this into an echo function and a validate function?

File: MetaGeneratorAbstractions/UserInput/SimpleValidation.cs [hunk 6]
```diff
@@ -1,10 +1,10 @@
-using System;
+﻿using System;
 
 namespace Messerli.MetaGeneratorAbstractions.UserInput
 {
     public class SimpleValidation : IValidation
     {
-        public SimpleValidation(Predicate<string> validation, string message)
```
Comment (tautropfli) [inline MetaGeneratorAbstractions/UserInput/SimpleValidation.cs:7]: I'd replace this with `Func`

File: Plugins/MetaGeneratorProjectPlugin/InputValidations.cs [hunk 7]
```diff
@@ -1,11 +1,19 @@
...
 
 namespace Messerli.MetaGeneratorProjectPlugin
 {
     internal static class InputValidations
     {
+        private const string MetaGeneratorSolution = "MetaGenerator.sln";
+
         public static IValidation MetaGeneratorSolutionExists
-            => new SimpleValidation(path => File.Exists(Path.Combine(path, "MetaGenerator.sln")), "No MetaGenerator.sln found at given location.");
+            => SimpleValidation.Create(MetaGeneratorSolutionInPath, "No MetaGenerator.sln found at given location.");
```
Comment (tautropfli) [inline Plugins/MetaGeneratorProjectPlugin/InputValidations.cs]:
```suggestion
            => SimpleValidation.Create(MetaGeneratorSolutionInPath, $"No {MetaGeneratorSolution} found at given location.");
```

**Review by tautropfli — CHANGES_REQUESTED** [review]

File: MetaGenerator/UserInput/ExistingPathRequester.cs (hunk 4 above)
Comment (FreeApophis) [reply MetaGenerator/UserInput/ExistingPathRequester.cs]: Having a party with his characterly friends!

File: FileManipulator/Project/MsBuild/CentralPackageVersionsManipulator.cs (hunk 1 above)
Comment (FreeApophis) [reply FileManipulator/Project/MsBuild/CentralPackageVersionsManipulator.cs]: Addressed in 9500c00e

File: MetaGenerator/TimeKeeper.cs (hunk 2 above)
Comment (FreeApophis) [reply MetaGenerator/TimeKeeper.cs]: Addressed in 9fc64c88dcd906d2d0de07d7ad6281413165b994

File: MetaGenerator/TimeKeeper.cs (hunk 3 above)
Comment (FreeApophis) [reply MetaGenerator/TimeKeeper.cs]: Addressed in 9fc64c88dcd906d2d0de07d7ad6281413165b994

File: MetaGenerator/UserInput/ExistingPathRequester.cs (hunk 4 above)
Comment (FreeApophis) [reply MetaGenerator/UserInput/ExistingPathRequester.cs]: Addressed in 5b45e69024b932fec60f71e87ab67e8baad6877f

File: MetaGenerator/UserInput/ValidatedUserInput.cs (hunk 5 above)
Comment (FreeApophis) [reply MetaGenerator/UserInput/ValidatedUserInput.cs]: Addressed in 531556de

File: MetaGeneratorAbstractions/UserInput/SimpleValidation.cs (hunk 6 above)
Comment (FreeApophis) [reply MetaGeneratorAbstractions/UserInput/SimpleValidation.cs:7]: Addressed in dd01b817b2813ce3db235726af887deb2cac54b7

File: MetaGenerator.Test/UserInput/TimeRequesterTest.cs [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+using Moq;
+using Xunit;
+
+namespace Messerli.MetaGenerator.Test.UserInput
+{
+    public class TimeRequesterTest
+    {
+        public TimeRequesterTest()
+        {
+            CultureInfo.CurrentCulture = new CultureInfo("de-CH", false);
```
Comment (tautropfli) [inline MetaGenerator.Test/UserInput/TimeRequesterTest.cs]: Why don't you use a `TemporaryCultureSwitch`?

Comment (FreeApophis) [reply MetaGenerator.Test/UserInput/TimeRequesterTest.cs]: Add and use TemporaryCultureSwich: 8f27d6a59a64bdc90f0b09f097d3f3e139a98379

**Review by tautropfli — APPROVED** [review]

**Pull request merged**